### ISSUE TITLE
Black Screen on Tianma Display, Poco F1

### ISSUE THREAD

**Issue by Gyurma1988** [Shubhamvis98/kali-pinephone#52]
If i flash a Tianma Boot image, i have a black screen. With EBBG boot image i have screen, but without touch. 

My Poco has a Tianma Panel.

**Comment by Shubhamvis98** [comment]
Hi @Gyurma1988,
On first boot the Droid juicer extracts the firmwares from vendor partitions. If the firmwares are missing the touch panel won't work.
Here's the workaround:
Try to reflash the rootfs and tianma boot.img
or
Repack the tianma boot.img with firmwares and try to boot again.

Sometimes you'll see black screen but try to poweroff and then poweron. We also face this issue sometimes but it works fine most of the time.

**Comment by Gyurma1988** [comment]
Hi, thanks for quick response.

I reflashed the boot and rootfs, but nothing. 

I turned the screen off and on, but some error.

How can i repack the tianma boot.img with firmwares?

**Comment by Shubhamvis98** [comment]
First, you have to extract/copy split firmware files from vendor/modem partitions then use pilsquasher to merge and create mbn files. Then unpack boot.img and place the mbn files in ramdisk and repack. You can use AIK from my repository to unpack/repack the boot.img

**Comment by Gyurma1988** [comment]
I'm afraid this is a bit over my head. I'm just a regular user, not a programmer. Thank you for trying to help me out.

**Comment by Shubhamvis98** [comment]
> I'm afraid this is a bit over my head. I'm just a regular user, not a programmer. Thank you for trying to help me out.

Yes, that's a time consuming task. Droid-juicer service is doing this stuff at first boot. Try flashing mobian first. If it's working then kali should work as well. Or I'll check with someone who has tianma variant.

**Comment by Gyurma1988** [comment]
I tried Mobian, but I'm still having the same problem.

**Comment by Shubhamvis98** [comment]
Them maybe you're doing something wrong. Can you show me the output of `cat /proc/cmdline` from android os.

**Comment by Gyurma1988** [comment]
Sure

![1726421578210](https://github.com/user-attachments/assets/037c0efa-78db-42a5-b17f-8611e235a567)

**Comment by Shubhamvis98** [comment]
It's tianma. Not sure then what the exact issue. I'll let you know if I get someone with tianma variant.

**Comment by Gyurma1988** [comment]
Ok, thank you very much.

**Comment by Gyurma1988** [comment]
Can you apply this patch to your build?

https://lwn.net/ml/linux-kernel/20201028221302.66583-1-kholk11@gmail.com/

It's from Postmarketos os.

**Comment by Gyurma1988** [comment]
![Screenshot_2024-09-17-19-45-33-189_com brave browser](https://github.com/user-attachments/assets/9a0d1256-f034-460a-998e-674b23e2ea7b)

**Comment by Shubhamvis98** [comment]
I hope this issue was fixed with the other two boot images.
